### DISCUSSION TITLE
BIP44 Identity Derivation

### DISCUSSION THREAD

**Discussion by ozwaldorf** [dfinity/sdk#2622] [Feature Requests]
So it was mentioned that DFX might be getting first class support for seedphrases. The current plan (as far as I know) is to do a simple implementation, where you can import a seedphrase, and derive/save the first secp identity from it. This would be matching the basic quill functionality. I think the support should go even further, to match the functionality of keysmith, where multiple of a seeds derivations can be used and or saved. This could be done in 2 ways: 

1) Static version: enter seed, can derive a certain subaccount and save it's pem
2) On the fly: store the encrypted seed, and be able to pass a derivation with a flag. By default, the first/0 derivation should be used. Maybe an optional alias could be used to have separate saved idents?

**Comment by viviveevee** [comment]
Thank you for the suggestion, @ozwaldorf! The support for seed phrases was implemented in #2499 by @adamspofford-dfinity.
I think the additional derivations could be useful, but unless there is more interest in this I don't think we would prioritize this right now. I'll re-address this within the next 2-3 weeks, and if you haven't heard back from me feel free to ping me again

**Reply by ozwaldorf** [reply]
Thanks for the response ! Here and there, when I have free time, I'm also exploring doing the feature myself. If I can successfully get it working, I'll open a PR

A side note: I was playing around with the implementation on the latest git, and unfortunately wasn't able to correctly derive the default principal id from a 12 word seed like keysmith and plug is able to. Maybe there is a padding error that's changing the derivation a little?

**Reply by jplevyak** [reply]
Odd, yes it seems that keysmith and quill the same identity from the same phrase, but dfx derives different principals (identities) for those phrases.

**Reply by jplevyak** [reply]
Also, quill adds the (unecessary) EC PARAMTERS block which we could remove.

**Reply by viviveevee** [reply]
Oof, in that case we have multiple problems still... there's also #1424

**Reply by jplevyak** [reply]
The quill generated identities do parse correctly BTW, returning a PRIVATE KEY.  However if the EC PARAMETERS block is included it parses but returns an EC PRIVATE KEY.

**Reply by adamspofford-dfinity** [reply]
You're right about the different derivation path, that was an oversight. The problems with what Quill emits I'm getting around to; sns-quill is receiving the change first, after which I'll make the same change upstream.

**Reply by viviveevee** [reply]
https://github.com/dfinity/sdk/pull/2672 latest dfx master should now produce the same identity as quill from the same seed phrase
